Volume/Networking/Network.swift

```diff
@@ -87,3 +87,30 @@ struct GraphQLPublisher<Query: GraphQLQuery>: Publisher {
         subscriber.receive(subscription: subscription)
     }
 }
+
+class NetworkState: ObservableObject {
+    @Published var networkScreenFailed: [Screen : Bool] = [:]
+
+    public enum Screen: CaseIterable {
+        case homeList, publicationList, bookmarksList
+    }
+
+    init() {
+        for screen in Screen.allCases {
+            networkScreenFailed[screen] = false
+        }
+    }
+
+    func determineState(screen: Screen, _ completion: Subscribers.Completion<WrappedGraphQLError>) {
+        if case let .failure(error) = completion {
+            networkScreenFailed[screen] = true
+            print(error.localizedDescription)
+        } else {
+            networkScreenFailed[screen] = false
+        }
+    }
+
+    func didNetworkFail(for screen: Screen) -> Bool {
```

Comment:
```suggestion
    func networkDidFail(for screen: Screen) -> Bool {
```

Maybe `with` or `on` rather than `for`, but that's a very stylistic choice.